lib/cog/nested_file.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Cog.NestedFile do
+  @moduledoc """
+  Stores an arbitrary String on the filesystem of the Cog host in a file
+  named by combining the provided key and extension. Files are written to
+  a directory hierarchy that is created by splitting the filename into
+  two character segments and joining up to three of those segments with a
+  provided list of base path segments.
+
+  Note: The key is sanitized to remove dangerous characters, but the
+  base paths and extension are not. Make sure not to use user entered
+  values for these unsafe arguments.
```

Comment:
Yeah, the key is the only thing that is user specified. That note is there for future-me who wants to reuse this for something else so that I don't make a mistake an let user-specified data into the namespace for instance.